include/odp/api/spec/queue.h

```diff
@@ -155,6 +158,50 @@ int odp_queue_context_set(odp_queue_t queue, void *context, uint32_t len);
  */
 void *odp_queue_context(odp_queue_t queue);
 
+/**
+ * Get a queue handle of an event aggregator associated with a queue
+ *
+ * Returns a queue handle that can be used to refer to an event aggregator
+ * associated with this queue. Unless otherwise noted, the returned queue
+ * handle can be used in all contexts where queue handles are used.
+ * In particular, the queue handle can be used in odp_queue_enq() to
+ * enqueue events through an event aggregator to the underlying queue.
+ * Similarly, the queue handle can be given to packet I/O, classifier,
+ * crypto and IPsec to have the generated events be enqueued by ODP through
+ * the event aggregator.
+ *
+ * This function does not create a new queue but merely returns a reference
+ * to an aggregator queue which has the same life time as the underlying
+ * queue. The underlying queue must not be destroyed as long as any of its
+ * aggregators is in use. The aggregator queue gets destroyed when the
+ * underlying queue gets destroyed. An aggregator queue handle must not
+ * be passed to odp_queue_destroy().
+ *
+ * The aggregator queues have the same enq_mode as the underlying queue.
+ *
+ * The returned queue handle cannot be used for dequeuing events. It must
+ * not be passed to odp_queue_deq() or similar.
+ *
+ * The returned queue handle must not be passed to any of the
+ * odp_queue_context* or odp_queue_sched* functions and is never returned
+ * by odp_queue_lookup().
```

Comment:
Perhaps the forbidden functions could be listed explicitly in the same manner than for example in `odp_init_global()`.

Reply:
I did not yet change this.

Reply:
Clarified in v5.